zebra-rpc/src/methods.rs

```diff
@@ -878,19 +879,17 @@ where
                 mempool::Response::Transactions(unmined_transactions) => {
                     if !unmined_transactions.is_empty() {
                         let tx = unmined_transactions[0].transaction.clone();
-                        return GetRawTransaction::from_transaction(tx, None, verbose != 0)
-                            .map_err(|error| Error {
-                                code: ErrorCode::ServerError(0),
-                                message: error.to_string(),
-                                data: None,
-                            });
+                        return Ok(GetRawTransaction::from_transaction(tx, None, 0, verbose));
                     }
                 }
                 _ => unreachable!("unmatched response to a transactionids request"),
             };
 
             // Now check the state
-            let request = zebra_state::ReadRequest::Transaction(txid);
+            let request = zebra_state::ReadRequest::Transaction {
+                hash: txid,
+                should_return_confirmations: true,
```

Comment:
Can we just always return the confirmations? Is the extra complexity of an additional argument and optional return value required?

Reply:
Yes, I wanted to avoid checking the tip height, but I don't think we should be adding boolean flags for what fields to include in state service responses, removed so that it always returns the confirmations in [`6cb2823`](https://github.com/ZcashFoundation/zebra/pull/6287/commits/6cb2823c015f866aecf139902ce6b192ced99a10).

Tangent: a framework for selecting response fields a caller is interested in would be nice for condensing state requests and responses.